spec.bs

```diff
@@ -2660,6 +2696,32 @@ CORP violation report=] algorithm, as leaving it unfenced may cause a privacy le
   [=automatic beacon data/once=] set to true.
 </div>
 
+<div algorithm=get-initiator-ancestor-root-origin>
```

Comment:
Let's add a note or an example to the bottom of this algorithm explaining what it actually does / what level of the tree it targets. From my reading, if `|sourceDocument|` is:

```
Main frame <-- origin A
  <iframe> <-- origin B
    <fencedframe> <-- origin C
      <iframe> <-- origin D
       <iframe> <-- origin E
          |sourceDocument|
```

... then this algorithm will return origin A, right?

And if `|sourceDocument|` is:

```
<fencedframe> <-- origin A
  <iframe> <-- origin B
    <fencedframe> <-- origin C
      {{sourceDocument}}
```

... it will return A in this case too, right?

So basically it returns the nearest fenced frame's unfenced parent's fenced parent's origin? I think?

----

Separate question: is there a guarantee that this will be called on a Document that is somewhere inside of a fenced frame tree?

Reply:
> So basically it returns the nearest fenced frame's unfenced parent's fenced parent's origin? I think?

Yes. The main use case for this is to get the origin of the fenced frame root whose `FencedFrameConfig` created the ad component that `|sourceDocument|` lives in. I've added a note to the algorithm.

> Separate question: is there a guarantee that this will be called on a Document that is somewhere inside of a fenced frame tree?

The returned value will only be used inside of a fenced frame tree, but since this is being set in the source snapshot params, it can be called on a Document outside of a fenced frame tree. The `If |navigable| is null, return null.` early return step will account for that.